src/main/java/org/logstash/beats/Server.java

```diff
@@ -43,16 +41,16 @@ public Server(String host, int p, int timeout, int threadCount) {
         port = p;
         clientInactivityTimeoutSeconds = timeout;
         beatsHeandlerThreadCount = threadCount;
-        workGroup = new NioEventLoopGroup();
     }
 
     public void enableSSL(SslSimpleBuilder builder) {
         sslBuilder = builder;
     }
 
     public Server listen() throws InterruptedException {
+        workGroup = new NioEventLoopGroup();
```

Comment:
@praseodym maybe we should make sure that if `workGroup` is `!= null` before this line, we will shutdown the "old" worker group first so we don't leak any theads here? I think it's not impossible to get to that situation since the reason we get to a reload is effectively always an `Exception` in the code that should shut this down right?

Reply:
Good point, I've added a null check. Reloads will mostly be caused by failing to _listen_, which leaves the worker group in a dead state (which is why reloading never worked), so I presume that the non-null shutdown will be mostly a no-op.